pyglmnet/utils.py

```diff
@@ -130,3 +135,22 @@ def set_log_level(verbose):
             raise ValueError('verbose must be of a valid type')
         verbose = logging_types[verbose]
     logger.setLevel(verbose)
+
+
+def return_verbose_iterable(data):
+    """
+    Wrap an iterable object with tqdm. If tqdm is not available
+    or if we did not set the appropriate log level, then we
+    fall back to the classical method
+    :param data: it can be a list or range() object
+    :param verbose: the verbosity level
+    :return: wrapped object
+    """
+    wrapped_data = data
+    if logger.getEffectiveLevel() == logging.INFO:
+        if not has_tqdm:
+            logging.info(
+                "tqdm library not found. Falling back to non-interactive.")
+        else:
+            wrapped_data = tqdm(data)
```

Comment:
does this work also on Jupyter notebook? have you checked? I remember having issues with tqdm on the notebook.

Reply:
Yes, it works nicely (see https://pypi.org/project/tqdm/#ipython-jupyter-integration). Honestly, when using cross-validation it can become a bit messy (especially if we throw too many `not converged` messages).